### ISSUE TITLE
Object detection: Tracking loss on test set, early stopping

### ISSUE THREAD

**Issue by thomasjanssens** [tensorflow/models#3593]
I'm interested in monitoring performance of a Faster RCNN object detection model on a custom dataset and stopping training before overfitting occurs. I've got two questions/feature requests related to this:
- Is it currently possible to evaluate the actual loss function on the test set, ideally every N (e.g. 1000) iterations?
- Is it possible to save the checkpoint that's the highest-scoring on some evaluation metric (e.g. test loss), or to do some other from of early stopping? 

It doesn't look like either of these things are addressed in the current documentation/implementation and will need some extra code, which seems weird, as I'd think both would be very nice to have when training a custom model. Is there any specific reason I'm missing why there doesn't seem to be a demand for these features?

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by harshini-gadige** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by paviddavid** [comment]
@thomasjanssens Have you solved this issue in the meantime? Any help is appreciated on this topic.

**Comment by huberemanuel** [comment]
Any updates on this?
